File: examples/native-host/ios/LynxNotificationsHostIntegration.swift
import Foundation

public enum LynxNotificationsHostIntegration {
  public static func install(
    moduleRegistrar: LynxNotificationsInstaller.ModuleRegistrar,
    methodAuthRegistrar: LynxNotificationsInstaller.MethodAuthRegistrar
  ) {
    let provider = FcmPushTokenProvider(tokenFetcher: {
      // Replace with FirebaseMessaging token retrieval.
      "replace-with-real-fcm-token"
    })

    let module = LynxNotificationsModule.createDefault(fcmProvider: provider)

    LynxNotificationsInstaller.install(
      moduleRegistrar: moduleRegistrar,
      methodAuthRegistrar: methodAuthRegistrar,
      module: module
    )
  }
}

File: examples/native-host/android/LynxNotificationsHostIntegration.java
package io.lynx.notifications.example;

import io.lynx.notifications.core.LynxNotificationsInstaller;
import io.lynx.notifications.core.LynxNotificationsModule;
import io.lynx.notifications.fcm.FcmPushTokenProvider;

/**
 * Host app wiring example.
 */
public final class LynxNotificationsHostIntegration {
  private LynxNotificationsHostIntegration() {}

  public static void install(
      LynxNotificationsInstaller.ModuleRegistrar moduleRegistrar,
      LynxNotificationsInstaller.AuthValidatorRegistrar authValidatorRegistrar
  ) {
    FcmPushTokenProvider provider = new FcmPushTokenProvider(() -> {
      // Replace with FirebaseMessaging.getInstance().getToken().await() or callback bridge.
      return "replace-with-real-fcm-token";
    });

    LynxNotificationsModule module = LynxNotificationsModule.createDefault(provider);

    LynxNotificationsInstaller.install(
        moduleRegistrar,
        authValidatorRegistrar,
        module
    );
  }
}
